---
title: "SimDatakumar"
output: html_document
---

Script and code follows "A step-by-step workflow for low-level analysis of single-cell RNA-seq data with Bioconductor". Aaron T. L. Lun1, Davis J. McCarthy2 and John C. Marioni. 

# Load and process data set: Kumar 2014 GSE60749-GPL13112
Note that the design is possibly completely confounded, as the conditions and batches are identical.
```{r load libraries, include=FALSE}
pdf("results/QC_data/QC_plots_simDataKumar.pdf")
# set seed
set.seed(1234)
#knittr options 
suppressPackageStartupMessages( library(MultiAssayExperiment) )
suppressPackageStartupMessages( library(scater) )
suppressPackageStartupMessages( library(dplyr) )
suppressPackageStartupMessages( library(tidyr) )
suppressPackageStartupMessages( library(plyr) )
suppressPackageStartupMessages( library(scran) )
suppressPackageStartupMessages( library(cowplot) )
suppressPackageStartupMessages( library(tibble) )
suppressPackageStartupMessages( library(splatter) )
```

## load multiassay dataset from IMLS conquer

```{r load dataset}
# load multiassay dataset from IMLS conquer
maex <- readRDS("data/GSE60749-GPL13112.rds")
# summary of data set
print(maex)
dim(assay(maex)) # 45686 x 268
# extract the gene-level TPM and the counts obtaines by scaling the TPMs
cts <- assays(experiments(maex)[["gene"]])[["count_lstpm"]]
tpms <- assays(experiments(maex)[["gene"]])[["TPM"]]
```
# extract the Phenotype data
There are three different subpopulations: knockout mouse and v6.5 mouse treated with serum or 2i.
```{r phentotypes, echo=TRUE}
# extract the phenotype data
phn <- colData(maex)

# extract the cell annotation to phenoid column and rename it 
phn[, "phenoid"] <- as.character(interaction(as.data.frame(phn[, c("source_name_ch1","characteristics_ch1.1")])))
phn[, "phenoid"] <-  revalue(phn[, "phenoid"] , c("Dgcr8 knockout mouse embryonic stem cells.culture conditions: serum+LIF"="Dgcr8 knockout mouse serum+LIF", "v6.5 mouse embryonic stem cells.culture conditions: 2i+LIF"="v6.5 mouse 2i+LIF","v6.5 mouse embryonic stem cells.culture conditions: serum+LIF"="v6.5 mouse serum+LIF"))
table(phn[, "phenoid"]) 
colData(maex) <- phn


```

# subset the dataset by the v6.5 mouse 2i+LIF cell population.
```{r subsetting cells, echo=TRUE }
# extract subpopulation names
maex.sub <- maex[,maex$phenoid == "v6.5 mouse 2i+LIF",]
cell.names <- assay(maex.sub) %>%colnames(sub.cts)
# subsetting
sub.cts <- cts[,cell.names]
dim(sub.cts)
```
# remove the ERCC spike inns
```{r remove spike inns, echo=TRUE }
is.spike <- grepl("^ERCC", rownames(sub.cts))
sub.cts <- sub.cts[ !is.spike,]

```
## Estimate parameters for simulation
The splatter package for  scRNA data simulation is used 
```{r estimate parameters, echo=TRUE}
params <- splatEstimate(sub.cts)

# have a look at the parameters
params
```

# Simulate new counts
Simulate SCEset dataset with 3 groups. Groups conatning of 100,200 and 500 cells. The probabilities for differential expression per group are 0.1,0.2 and 0.4.
```{r simulate counts, echo=TRUE}
# use splatSimulate to simulate counts

sceset <- splatSimulate(params=params,   batchCells = 500,
                        group.prob = c(0.1, 0.15, 0.5, 0.25), 
                        method = "groups",de.prob=c(0.05,0.1,0.2,0.4),
                        verbose = FALSE)
```

throw away features which are not expressed
```{r reduce expression matrix, throwing away features that are not expressed. noninformative features}
keep_features <- rowSums(counts(sceset) > 0) >= 1
table(keep_features)
sceset<- sceset[keep_features, ]
dim(sceset)
```

log transform counts
```{r log transform}

logcounts(sceset) <- log2(calculateCPM(sceset, 
                                            use.size.factors = FALSE) + 1)
```


#  add cellgroup labels
```{r populate phenoid}
#  store group under phenoid
sceset$phenoid <- colData(sceset)$Group
```
# tSNE and PCA plots by cellgroup

```{r plots}
plotTSNE(sceset, exprs_values = "counts", colour_by="phenoid", perplexity=10)
plotPCA(sceset, exprs_values = "counts", colour_by="phenoid")

```

# save counts
```{r save raw counts}
res <- sceset
save(res,file = "data/sceset_raw_simDataKumar.rda")
```


# Compute some QC metrics and set ERCc as spike inn controls

```{r QC}
sceset <- calculateQCMetrics(sceset )

```

# Quality control on cells
The number of expressed genes shows a bimodal distribution. Use log transforamtion for a shift in lower libsizes. 
```{r  histogramms }
size.med <- median(log10(sceset$total_counts), na.rm = TRUE)
size.mad <- mad(log10(sceset$total_counts), center = size.med, na.rm = TRUE)
lower.limit.size <- 10^(size.med - 3 * size.mad)/1e6
    
feature.med <- median(log10(sceset$total_features), na.rm = TRUE)
feature.mad <- mad(log10(sceset$total_features), center = size.med, na.rm = TRUE)
lower.limit.feat <- 10^(feature.med - 3 * feature.mad  )


pdf("results/QC_data/hist_simDataKumar.pdf")

par(mfrow=c(1,2))
hist(sceset$total_counts/1e6, xlab="Library sizes ", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
abline(v=lower.limit.size,col=2, lty=2 )
hist(sceset$total_features, xlab="Number of expressed genes", main="", 
    breaks=20, col="grey80", ylab="Number of cells")
abline(v=lower.limit.feat, col=2, lty=2 )
dev.off()
    
```

# Quality control using PCA and highest expressed genes
PCA plot based the quality metrics for each cell, e.g., the total number of reads, the total number of features and the proportion of mitochondrial or spike-in reads. 
```{r  Quality control using PCA }
p1 <- scater::plotPCA(sceset, exprs_values="logcounts", selected_variables="pdata",colour_by="phenoid")
p2 <- scater::plotQC(sceset, type = "find-pcs", variable = "total_features")

```

# Filter dataset gene wise and by librarysize 
Remove cells with log-library sizes that are more than 3 median absolute deviations (MADs) below the median log-library size. 21 cells were dropped according to this criteria, regarding the untransformed count data. 

Keep only those features that are observed in at least 1 cell. 435 features are filtered out. The original counts are used.


```{r library size and total features }

libsize.drop <- isOutlier(sceset$total_counts, nmads=3, type="lower", log=TRUE)
feature.drop <- isOutlier(sceset$total_features, nmads=3, type="lower", log=TRUE)
table(libsize.drop,feature.drop)

```

Total 4 cells were filtered out.  
```{r overview filtered cells }

sceset.red <- sceset[,!(libsize.drop | feature.drop )]
data.frame(ByLibSize=sum(libsize.drop), ByFeature=sum(feature.drop), Remaining=ncol(sceset.red))

```

Plotting the 50 highest expressed genes. The profile should look relatively "flat".
```{r  highest expressed genes}
p3 <- plotQC(sceset.red, type = "highest-expression", n=50 )

```

```{r mean freq expression}
plotQC(sceset.red, type = "exprs-freq-vs-mean")
```

# Filtering the datasets based on the features. Here genes with low abundance are filtered out. We have to sets of features one is based on the mean expression and one on a expression in minium n cells.
```{r  Filtering out low-abundance genes }
ave.counts <- rowMeans(counts(sceset.red))
keep <- ave.counts >= 1
sum(keep)
ave.counts<- as.data.frame(log10(ave.counts))

p4 <- ggplot(ave.counts, aes(ave.counts))+geom_histogram(binwidth = 0.5)+geom_vline(aes(xintercept=log10(1), color=1))+guides(color = "none")+labs( x = expression(Log[10]~"average count"), y = "frequency")
sceset.red <- sceset.red[keep, ]

```



Normalize data according to Lun etal. 2016 using pooled counts to estimate size factors. The size factors are positevely correlated with the library sizes for all cells. We see two groups, possibly coming from the two cell  populations.
```{r normalize through pool-based size factors }
sceset <- computeSumFactors(sceset, sizes=c(20, 40, 60, 80))
summary(sizeFactors(sceset))

sceset.red <- computeSumFactors(sceset.red, sizes=c(20, 40, 60, 80))
summary(sizeFactors(sceset.red))

plot(sizeFactors(sceset), sceset$total_counts/1e6, log="xy",
    ylab="Library size (millions)", xlab="Size factor")
plot(sizeFactors(sceset.red), sceset.red$total_counts/1e6, log="xy",
    ylab="Library size (millions)", xlab="Size factor")
```


The counts are normalized using log2 transformation using the edgeR::calcNormFactors function.
```{r normalization }
sceset <- normaliseExprs(sceset)
sceset.red <- normaliseExprs(sceset.red)
```

# Save the normalized and cell filtered datasets
```{r save data }
res <- sceset
save(res,file = "data/sceset_org_simDataKumar.rda")

res <- sceset.red
save(res,file = "data/sceset_red_simDataKumar.rda")

```


# Identifying HVGs from the normalized log-expression

```{r High variable genes ,  eval=FALSE}

var.fit <- trendVar(sceset.red, trend="loess", use.spikes=FALSE, span=0.2) 
var.out <- decomposeVar(sceset.red, var.fit)


```


```{r High variable genes using the variance of the log-expression,  eval=FALSE }

plot(var.out$mean, var.out$total, pch=16, cex=0.6, xlab="Mean log-expression",
    ylab="Variance of log-expression")
o <- order(var.out$mean)
lines(var.out$mean[o], var.out$tech[o], col="dodgerblue", lwd=2)
cur.spike <- isSpike(sceset.red)
points(var.out$mean[cur.spike], var.out$total[cur.spike], col="red", pch=16)


hvg.out <- var.out[which(var.out$FDR <= 0.05 & var.out$bio >= 0.5),]
hvg.out <- hvg.out[order(hvg.out$bio, decreasing=TRUE),]
nrow(hvg.out)
head(hvg.out)
plotExpression(sceset.red, rownames(hvg.out)[1:10])

```
# select only high variable genes and save sceset
```{r hvg, , eval=FALSE}

sceset.hvg <-sceset.red[ rownames(hvg.out), ]
dim(sceset.hvg)

res <- sceset.hvg
save(res,file = "data/sceset_hvg_simDataKumar.rda")


```
## Generate different plots for the summary
Plot the proportion of explained variances
```{r explained variance} 
expl_vars <- c( "Group","log10_total_counts", "log10_total_features", 
               "pct_counts_top_200_features")
p5 <- plotQC(sceset.red, type = "explanatory-variables", variables = expl_vars)
```
plot tSNE represenatation
```{r tSNE}
p6 <- plotTSNE(sceset.red, exprs_values="logcounts", perplexity=5, colour_by="total_features",size_by="total_counts",return_SCESet=FALSE) + ggtitle("tSNE plot")
```


Plot a summary of the QC
```{r create QC plots} 


plot.qc <- plot_grid(p1,p2,p3,p4,p5,p6, ncol = 2, labels="auto")

save_plot("results/QC_data/qc_summary_simDataKumar.png", plot.qc,
          ncol = 2, # we're saving a grid plot of 2 columns
          nrow = 2 # and 2 rows
          )

dev.off()
```


